dali/kernels/reduce/reduce_axes_gpu_impl.cuh

```diff
@@ -91,6 +109,33 @@ __device__ void ReduceNone(Out *out, const In *in, int64_t n,
   }
 }
 
+/**
+ * @brief This function is used when the reduction is no-op (reduced extent is 1),
+ *        but the reduced axis is not inner, so we need to account for complex addressing
+ *        of the preprocessor.
+ *
+ * This function will apply preprocessing and postprocessing, but will not do
+ * any actual reduction.
+ *
+ * @param pre_bank      preprocessor bank, providing possibly distinct procssing
+ *                      per output sample
+ * @param post          posptprocessing unary functor
+ */
+template <typename Out, typename In, typename PreprocessorBank, typename Postprocessor>
```

Comment:
This is the only device function that's actually new. The previous "ReduceNone" function only worked with the innermost dimension.